synapse/python_dependencies.py

```diff
@@ -15,175 +15,109 @@
 # limitations under the License.
 
 import logging
-from distutils.version import LooseVersion
 
-logger = logging.getLogger(__name__)
+from pkg_resources import DistributionNotFound, VersionConflict, get_distribution
 
-# this dict maps from python package name to a list of modules we expect it to
-# provide.
-#
-# the key is a "requirement specifier", as used as a parameter to `pip
-# install`[1], or an `install_requires` argument to `setuptools.setup` [2].
-#
-# the value is a sequence of strings; each entry should be the name of the
-# python module, optionally followed by a version assertion which can be either
-# ">=<ver>" or "==<ver>".
-#
-# [1] https://pip.pypa.io/en/stable/reference/pip_install/#requirement-specifiers.
```

Comment:
suggestion: it'd be nice to keep this link so I don't have to go and look it up.